src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -300,154 +217,75 @@ private static bool IsInternalsVisibleToAttribute(AttributeData attr)
+        private static HashSet<string> GetInternalsVisibleToSet(IAssemblySymbol assembly)
         {
-            var internalsVisibleToMap = new Lazy<HashSet<string>>(() =>
+            var set = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var attr in assembly.GetAttributes().Where(IsInternalsVisibleToAttribute))
             {
-                var map = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-                foreach (var attr in assembly.GetAttributes().Where(IsInternalsVisibleToAttribute))
+                var typeNameConstant = attr.ConstructorArguments.FirstOrDefault();
+                if (typeNameConstant.Type == null ||
+                    typeNameConstant.Type.SpecialType != SpecialType.System_String ||
+                    !(typeNameConstant.Value is string value))
                 {
-                    var typeNameConstant = attr.ConstructorArguments.FirstOrDefault();
-                    if (typeNameConstant.Type == null ||
-                        typeNameConstant.Type.SpecialType != SpecialType.System_String ||
-                        !(typeNameConstant.Value is string value))
-                    {
-                        continue;
-                    }
+                    continue;
+                }
 
-                    var commaIndex = value.IndexOf(',');
-                    var assemblyName = commaIndex >= 0 ? value.Substring(0, commaIndex).Trim() : value;
+                var commaIndex = value.IndexOf(',');
+                var assemblyName = commaIndex >= 0 ? value.Substring(0, commaIndex).Trim() : value;
 
-                    map.Add(assemblyName);
-                }
+                set.Add(assemblyName);
+            }
 
-                return map;
-            }, isThreadSafe: true);
-            return internalsVisibleToMap;
+            return set;
         }
 
         private static bool HasReferenceTo(
-            IAssemblySymbol containingAssembly,
-            Project? sourceProject,
+            (IAssemblySymbol assembly, Project? sourceProject) symbolOrigination,
             Project project,
             CancellationToken cancellationToken)
         {
-            if (containingAssembly == null)
-            {
-                throw new ArgumentNullException(nameof(containingAssembly));
-            }
-
-            if (project == null)
-            {
-                throw new ArgumentNullException(nameof(project));
-            }
-
-            if (sourceProject != null)
-            {
-                // most of time, compilation should be already there
-                return project.ProjectReferences.Any(p => p.ProjectId == sourceProject.Id);
-            }
-
-            return project.HasReferenceToAssembly(containingAssembly, cancellationToken);
-        }
+            Contract.ThrowIfNull(symbolOrigination.assembly);
+            Contract.ThrowIfNull(project);
```

Comment:
We're not trusting nullable reference types here?

Reply:
i like contracts :)